### ISSUE TITLE
Improve Performance of Fetching Descriptor file via TRS

### ISSUE THREAD

**Issue by coverbeck** [dockstore/dockstore#4480]
**Is your feature request related to a problem? Please describe.**
The performance can be a little slow, at least locally, to fetch a single descriptor file.

Fetching one file from one particular workflow version results in 241 JDBC statements. This workflow has 572 versions:

```
curl -s -X GET "http://localhost:8080/api/ga4gh/v1/tools/%23workflow%2Fgithub.com%2Fbroadinstitute%2Fgatk%2Fcnv_somatic_pair_workflow/versions" -H  "accept: application/json" | jq '. | length'
572
```

**Describe the solution you'd like**

The [current code](https://github.com/dockstore/dockstore/blob/cc2a1152102823bac1d0453dbb672ab8b4b53030/dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java#L508) loads up an entry and fetches all of its versions. It then looks in Java for the matching version to find the correct source file. That means it's fetching all the versions (572 in my example above) only to discard all but one.

Instead, the code could just fetch the version or maybe even the source files directly from the DB, although it's slightly more complicated than that; see surrounding code.

**Describe alternatives you've considered**
Leave it. Seems to perform much better in a deployed env than in my local setup.

**Additional context**
This is one of our heavily hit endpoints, when Terra is executing a workflow across many containers. On workflows without many versions it's very fast, the one the one in my example it takes a couple of seconds, whereas its under 100ms for simple descriptors.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/DOCK-1921)
┆fixVersions: Dockstore 1.13
┆friendlyId: DOCK-1921
┆sprint: SEAB 83- Luca
┆taskType: Story

**Comment by denis-yuen** [comment]
Makes sense to transform our frequently used endpoints from object manipulation to JPQL queries

**Comment by denis-yuen** [comment]
No, this is for a single file (which can and should be optimised, possibly with JPQL).
This is different from the (multiple) tools call being optimized in https://github.com/dockstore/dockstore/issues/2125
